### PR TITLE
[DOCS] Updating getting started guide

### PR DESCRIPTION
**Pull request by ChrisChinchilla**
After a lot of work, this closes https://github.com/ethereum/go-ethereum/issues/20396

The steps are somewhat brittle, and don't know if there are better ways to get clef and geth speaking with each other that I missed, in which case there are probably other docs that need polish to make that clearer.

If people could take a look at the actual steps, and once they are polished, I'll take another pass at the text and add a few more images

### CODE REVIEW

**Comment by fjl** [comment]
I really like this guide, but it kind of sucks that `geth account new` is needed to make `clef` work. It should really be possible to run `clef` on an empty keystore.

**Comment by holiman** [comment]
@fjl yes unfortunately. But it's silly, so PTAL:https://github.com/ethereum/go-ethereum/pull/20782

**Comment by ChrisChinchilla** [comment]
So… @fjl and @holiman shall we merge this for now, or wait for that PR? I mean, it's fiddle, but not terrible, and I can update the guide to mention that the change is in progress

**Comment by fjl** [comment]
We can go ahead with this when #20782 is released. That will happen next week.

**Comment by fjl** [comment]
The `clef newaccount` feature is now available in a release.

**Comment by ChrisChinchilla** [comment]
Nice @fjl I updated, makes it so much clearer!

I think this also means the clef docs need updating, can take on that.

File: docs/_getting-started/index.md [hunk 1]
```diff
@@ -4,89 +4,202 @@ permalink: docs/getting-started
...
+
+RPC allows remote applications to access your node but has limitations and security
+considerations, and by default only allows access to methods in the `eth` and `shh`
+namespaces. Find out how to override this setting [in the RPC docs](rpc/server#http-server). 
+
+## Using IPC
+
+### Connect to console
+
+You can connect to the IPC console in two ways. First, when you start the node:
```
Comment (fjl) [inline docs/_getting-started/index.md:108]: I think we shouldn't even have the `geth console` way in the guide anymore. It's kind-of redundant and just means more flags need to be passed. `geth attach` is better.

File: docs/_getting-started/index.md [hunk 2]
```diff
@@ -4,89 +4,202 @@ permalink: docs/getting-started
...
+### Check account balance
+
+```shell
+curl -X POST http://<GETH_IP_ADDRESS>:8545 --data \
+    '{"jsonrpc":"2.0",
+    "method":"eth_getBalance",
+    "params":["<ADDRESS_1>","latest"],
+    "id":1}' \
+    -H "Content-Type:application/json"
+```
```
Comment (fjl) [inline docs/_getting-started/index.md:174]: These examples would look better if you put the RPC payload on a single line, e.g.

```
curl -X POST http://<GETH_IP_ADDRESS>:8545 -H "Content-Type:application/json" --data \
    '{"jsonrpc": "2.0", "id": 1, "method": "eth_getBalance", "params": ["<ADDRESS_1>", "latest"]}'
```

**Comment by fjl** [comment]
Looks very good now. One thing that bugged me a bit is that the interaction gifs are playing quite quickly and it's hard to follow the output. Maybe better to use asciinema for this.

**Comment by ChrisChinchilla** [comment]
@fjl Comments addressed, I actually used something very similar to asciinema (terminalizer), and intentionally made them faster, it's a YAML config that you then render to a gif, so I can adjust the speed and re-render, will do that now.

**Comment by ChrisChinchilla** [comment]
@ChrisChinchilla Images updated

**Comment by ChrisChinchilla** [comment]
Further thoughts @fjl ?

**Comment by fjl** [comment]
Sorry, I forgot about this again. Thank you for the update!
